### PR TITLE
[GEP-26] CredentialsBinding validation via admission webhook

### PR DESCRIPTION
**Pull request by dimityrmirchev**
**How to categorize this PR?**

/area security ipcei
/kind enhancement
/platform aws
/label ipcei/workload-identity

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/9586

**Special notes for your reviewer**:

**Release note**:

```feature user
The admission webhook now validates `CredentialsBinding`s.
```

### CODE REVIEW

**Comment by dimityrmirchev** [comment]
cc @vpnachev

**Review by vpnachev — CHANGES_REQUESTED** [review]
Some nits and one issue that I found, the admission controller fails to start up because the scheme is missing entry for the CredentialsBinding resource. 

You have to add 

```go
github.com/gardener/gardener/pkg/apis/security/install.Install(mgr.GetScheme())
```

somewhere here https://github.com/gardener/gardener-extension-provider-aws/blob/5a0ca81b53918509d147b0e0c141480fdc4266a3/cmd/gardener-extension-admission-aws/app/app.go#L120

File: pkg/admission/validator/credentialsbinding.go
Comment (dimityrmirchev) [reply pkg/admission/validator/credentialsbinding.go:44]: Yes, I think that we can hold off until that PR is merged.

**Comment by dimityrmirchev** [comment]
Thanks for reviewing, I will also address your feedback in the other PRs that I opened for other provider extensions. Meanwhile I will move their status to draft until this one is finalized.

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**
